src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+Imports System.ComponentModel
+Imports System.Linq
+Imports System.Runtime.InteropServices
+Imports Xunit
+' Do not Imports System.IO
+
+
+Namespace Microsoft.VisualBasic.Tests
+    Public NotInheritable Class FileIOTests
+        ReadOnly DestData() As Char = {"x"c, "X"c, "y"c}
+
+        ReadOnly SourceData() As Char = {"a"c, "A"c, "b"c}
+
+        Sub New()
+        End Sub
+
+        Private Function CreateTestFile(TestBase As FileIOTestBase, ByVal TestData() As Char, Optional TestFileNameWithPath As String = "") As String
+            If TestFileNameWithPath.Length = 0 Then
+                TestFileNameWithPath = TestBase.GetTestFilePath()
+            Else
+                Assert.True(IO.Path.IsPathRooted(TestFileNameWithPath))
+                Assert.True(TestFileNameWithPath.StartsWith(TestBase.TestDirectory))
+            End If
+            ' Write and copy file
+            Using sourceStream As New IO.StreamWriter(IO.File.Create(TestFileNameWithPath))
+                sourceStream.Write(TestData, 0, TestData.Length)
+            End Using
+
+            Return TestFileNameWithPath
+        End Function
+
+        <Fact>
+        Public Sub CombinePathTest_BadBaseDirectory_RelativePath()
+            Assert.Throws(Of ArgumentNullException)(Function() FileSystem.CombinePath("", "Test2"))
+            Assert.Equal(FileSystem.CombinePath("C:\", ""), "C:\")
+        End Sub
+
+        <Fact>
+        Public Sub CombinePathTest_BaseDirectory_RelativePath()
+            Assert.Equal(FileSystem.CombinePath("C:\", "Test2"), IO.Path.Combine("C:\", "Test2"))
+        End Sub
+        <Fact>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName()
+            Dim TestBase As New FileIOTestBase
+            Dim FullPathToSourceDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "SourceDirectory")
+            Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "TargetDirectory")
+            IO.Directory.CreateDirectory(FullPathToSourceDirectory)
+            For i As Integer = 0 To 5
+                CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToSourceDirectory, $"NewFile{i}"))
+            Next
+            FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory)
+            Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+            For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                ' Ensure copy transferred written data
+                Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
+                    Dim ReadData(SourceData.Length - 1) As Char
+                    stream.Read(ReadData, 0, SourceData.Length)
+                    Assert.Equal(SourceData, ReadData)
+                End Using
+            Next
+            IO.Directory.Delete(FullPathToTargetDirectory, recursive:=True)
```

Comment:
can split the failure part into its own test

Reply:
I choose not to do that to avoid duplicating all the setup. I can make the change if it is required.

Reply:
we can move the setup into a common  function.
we try  to avoid testing multiple behaviours in a single test.